src/SignUp/SignUp.js

```diff
@@ -0,0 +1,60 @@
+import { gql, useMutation } from '@apollo/client';
+import React, { useState } from 'react'
+import { useHistory } from 'react-router-dom'
+
+export const USER_SIGNUP = gql`
+  mutation userSignup($username: String!, $password: String!){
+    createUser(input: {
+      username: $username
+      password: $password
+    })
+    {
+      user{
+        id
+        username
+      }
+    }
+  }
+`;
+
+export function SignUp() {
+  let [error, setError] = useState(false)
+  let input = {
+    password: '',
+    username: ''
+  }
+  let history = useHistory();
+  let [signUpUser, { data }] = useMutation(USER_SIGNUP);
+
+  let handleInput = (e) => {
+    input[e.target.name] = e.target.value
+  }
+
+  let signUp = (e) => {
+    let {username, password} = input
+    e.preventDefault()
+    signUpUser({
+      variables: {
+        username,
+        password
+      }
+    }).then( response => {
+      history.push('/')
```

Comment:
I am still looking into a way to get the NavLinks to go '/' instead of '/home' not sure if this will be an issue here

Reply:
Good heads up. Hopefully it won't be too troublesome either way!